src/DynamoApplications/StartupUtils.cs

```diff
@@ -228,5 +228,65 @@ public static string SetLocale(CommandLineArguments cmdLineArgs)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Handler for an assembly load event into a host's appdomain - we need to make sure
+        /// that another addin or package has not loadead another version of a .dll that we require.
+        /// If this happens Dynamo will most likely crash. We should alert the user they
+        /// have an incompatible addin installed.
+        /// 
+        /// </summary>
+        /// <param name="assembly"></param>
+        /// <param name="assemblyNamesToIgnore"></param>
+        /// <returns></returns>
+        public static List<Exception> AppDomainHasMismatchedReferences(Assembly assembly, String[] assemblyNamesToIgnore)
```

Comment:
Could you alter the function name, as xxxhasxxx sounds more like returning a boolean? Maybe also wrap this with something like GetPreloadExceptions() so that it would be more straight-forward when called in client code. Something to think about :)